packages/ui-react/src/containers/AccountModal/forms/Checkout.tsx

```diff
@@ -30,6 +33,10 @@ const Checkout = () => {
   const welcomeUrl = modalURLFromLocation(location, 'welcome');
   const closeModalUrl = modalURLFromLocation(location, null);
 
+  const recaptchaRef = useRef<ReCAPTCHA>(null);
+  const captchaSiteKey = useConfigStore(({ config }) => (config.custom?.captchaSiteKey ? (config.custom?.captchaSiteKey as string) : undefined));
+  const getCaptchaValue = useEventCallback(async () => (captchaSiteKey ? (await recaptchaRef.current?.executeAsync()) || undefined : undefined));
+
```

Comment:
Maybe we could have a custom hook here and also reuse it for registration?

Reply:
Not a bad idea indeed. Would you say we also forward the ref to the custom hook? 
```javascript
const { recaptchaRef, captchaSiteKey } = useRecaptcha();
```

Reply:
Yeah, something like this.

Reply:
I've added `useRecaptcha()`